hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/IncrementalContainerReportHandler.java

```diff
@@ -91,6 +91,8 @@ public void onMessage(final IncrementalContainerReportFromDatanode report,
             if (!replicaProto.getState().equals(
                 ContainerReplicaProto.State.DELETED)) {
               nodeManager.addContainer(dd, id);
+            } else {
```

Comment:
Should we check for `replicaProto.getState().equals(ContainerReplicaProto.State.DELETED)` to remove the container?

Reply:
I did originally have an `else(DELETED)` statement, but then I realized the IF part is for "not deleted", so else can only be deleted. It may be safer for the future to make it an `else if`.

Reply:
I switched the order around to make it more clear what is going on.